planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/ConfiguredMapMain.java

```diff
@@ -0,0 +1,79 @@
+package com.onthegomap.planetiler.custommap;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.onthegomap.planetiler.Planetiler;
+import com.onthegomap.planetiler.config.Arguments;
+import com.onthegomap.planetiler.custommap.configschema.DataSource;
+import com.onthegomap.planetiler.custommap.configschema.DataSourceType;
+import com.onthegomap.planetiler.custommap.configschema.SchemaConfig;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import org.yaml.snakeyaml.Yaml;
+
+public class ConfiguredMapMain {
+
+  private static final Yaml yaml = new Yaml();
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  /*
+   * Main entrypoint
+   */
+  public static void main(String[] args) throws Exception {
+    run(Arguments.fromArgsOrConfigFile(args));
+  }
+
+  private static void run(Arguments args) throws Exception {
+    var dataDir = Path.of("data");
+    var sourcesDir = dataDir.resolve("sources");
+
+    var schemaFile = args.inputFile(
+      "schema",
+      "Location of YML-format schema definition file");
+
+    var config = loadConfig(schemaFile);
+
+    var planetiler = Planetiler.create(args)
+      .setProfile(new ConfiguredProfile(config));
+
+    var sources = config.getSources();
+    for (var source : sources.entrySet()) {
+      configureSource(planetiler, sourcesDir, source.getKey(), source.getValue());
+    }
+
+    planetiler.overwriteOutput("mbtiles", Path.of("data", "output.mbtiles"))
+      .run();
+  }
+
+  static SchemaConfig loadConfig(Path schemaFile) throws IOException {
+    Map<String, Object> parsed = yaml.load(Files.newInputStream(schemaFile));
+    return mapper.convertValue(parsed, SchemaConfig.class);
+  }
+
+  private static void configureSource(Planetiler planetiler, Path sourcesDir, String sourceName, DataSource source)
+    throws URISyntaxException {
+
+    DataSourceType sourceType = source.type();
+
+    switch (sourceType) {
+      case OSM:
```

Comment:
minor: we should try to use the newer switch expression format now that it's available:

```java
switch (sourceType) {
  case OSM -> {
    do stuff, no return needed!
  }
  case SHAPEFILE -> {
    do stuff, no return
  }
  default -> ...
}
```